### DISCUSSION TITLE
License Consideration when Using Nominatim for Only Producing DB

### DISCUSSION THREAD

**Discussion by panjiesw** [osm-search/Nominatim#3256] [Q&A]
Hello, I know this might not be the best place to ask about License policies but still wondering what others' opinions, especially maintainers, regarding this are.

I want to use Nominatim but only for the purpose of producing the database snapshot (I'm thinking of doing this either weekly or monthly). The building process will be in my own infrastructure.

In another process, I would like to then import the snapshot into separate PostgreSQL DB and consume it in my own App Server running also in my own infrastructure, with specific and currently private query / code logic.

My question is, since Nominatim is under GPL v2 and I plan on using the DB output by Nominatim import process, do I have to license my App Server code under the same GPL v2 and open it?

**Comment by lonvia** [comment]
The GPL is only relevant, if you extend the code of Nominatim itself. This is not the case here. To give you an analogous example: when you write a document in LibreOffice, the document itself can be in an license and you are of course free to open it in MS Office even though it is a closed-source product. Using data in different products does not trigger share-alike.

What is relevant to your case: the database you produce here consists of OpenStreetMap data. This comes with the ODbL license, which is also share-alike and requires attribution. Again, this will have no direct relevance for your App Server code but it does have a relevance for when you use the data. If you have a search box in the App, you must mention OSM as a source. If you use the database to mix it with other data, then the other data may be effected by the share-alike clause. Please read our [community guideline on licensing with geocoding results](https://osmfoundation.org/wiki/Licence/Community_Guidelines/Geocoding_-_Guideline) for more information.

For more detailed legal advice, you have to contact a lawyer, I'm afraid.

**Reply by panjiesw** [reply]
Thank you for response @lonvia! Now I understand  a bit more and indeed more detailed advice should come from appropriate entities such as lawyers. 

If I may ask another opinion from you regarding the share-alike clause: If I have another data source (say, from a commercial data subscription) stored in a physically and completely separated DB from the OSM DB, but mix the search result from both DB, does it fall into share-alike clause? It's a bit like the [Using OSM-based Geocoding Results together with non-OSM Geocoding Results](https://osmfoundation.org/wiki/Licence/Community_Guidelines/Geocoding_-_Guideline#Using_OSM-based_Geocoding_Results_together_with_non-OSM_Geocoding_Results.) example, but with additional case of mixed result.
